Allow docker containers to verify tokens from keycloak

Matt-Guest opened the pull request:
The default host name in keycloak is "localhost". This allows only
requests that have localhost in them to verify any tokens.

Running a curl command from a docker container will result in the
following error;

{"error":"invalid_token","error_description":"Token invalid: Invalid token issuer. Expected 'http://keycloak:8080/auth/realms/development', but was 'http://localhost:8180/auth/realms/development'"}

To get around this I have set the hostname to be "keycloak" and
configured the port to be 8180 (As the port must match as well!)

The biggest change in the PR will be the keycloak port number which will
be 8180 within docker. I can't see this affecting other teams as
we seem to be the first team to run into this issue when verifying tokens within docker.

<!-- You can erase any questions or checklists in this template that are not applicable. -->

* **What kind of change does this PR introduce (Bug fix, feature, docs update, ...)?**
  Bug fix when verifying tokens in a docker container

* **What is the current behavior?**
  Only works outside docker

* **What is the new behavior (if this is a feature change)?**
  Works inside docker and outside docker

* **Does this PR introduce a breaking change? If so, what actions will users need to take in order to regain compatibility?**
 keycloak port within docker has changed to 8180 as it must match inside and outside of docker.

## Checklists

### All submissions

* [x] Have you followed the guidelines in our [Contributing](CONTRIBUTING.md) document?
* [x] Is this PR targeting the `develop` branch, and the branch rebased with commits squashed if needed?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

<!-- You can erase the following part of this template if it's not applicable to your Pull Request. -->

### New feature and bug fix submissions

* [x] Has your submission been tested locally?
* [x] Has documentation such as the [README](/README.md) been updated if necessary?
* [x] Have you linted your new code (using rubocop and markdownlint), and are not introducing any new offenses?

Matt-Guest commented:
You can still access keycloak via localhost outside docker

sdchapman commented:
Could you target develop branch please?

sdchapman commented on scripts/docker/auth/docker-compose-fragment.3.7.yml:
```diff
@@ -10,6 +10,6 @@ services:
     build: ../scripts/docker/auth/keycloak
     ports:
-      - "8180:8080"
+      - "127.0.0.1:8180:8180"
```
Is 127.0.0.1 needed here?

Matt-Guest replied:
Good spot, no its not needed.

sdchapman commented on scripts/docker/auth/docker-compose-fragment.yml:
```diff
@@ -10,6 +10,6 @@ services:
     build: ../scripts/docker/auth/keycloak
     ports:
-      - "8180:8080"
+      - "127.0.0.1:8180:8180"
```
Is 127.0.0.1 needed here?

Matt-Guest replied:
I have removed the IP address here too

Matt-Guest commented:
I have changed the target to develop and have addressed all review comments.

@sdchapman Can you please have another look when you have some free time?

Thanks
